Source/WebCore/loader/cache/CachedResourceLoader.cpp

```diff
@@ -1024,6 +1049,12 @@ ResourceErrorOr<CachedResourceHandle<CachedResource>> CachedResourceLoader::requ
         return makeUnexpected(ResourceError { errorDomainWebKitInternal, 0, url, "Not allowed to request resource"_s, ResourceError::Type::AccessControl });
     }
 
+    if (RefPtr document = this->document()) {
+        bool shouldUpgradeMixedContent = MixedContentChecker::shouldUpgradeInsecureContent(frame, isUpgradableTypeFromResourceType(type), url, request.options().mode, request.options().destination, request.options().initiator);
```

Comment:
This seems redundant with the upgradeInsecureRequestIfNeeded call 10 lines above.
Why also doing it after canRequest call?

Reply:
The comment above canRequest said it is handling steps 2 and 4 in the main fetch spec, and upgrading mixed content is step 6. However, now I see that the comment above canRequest is stale and it is actually implementing step 4, 7, as 12? I'll need to look closer at that later, but I believe we can combine the two calls to upgradeInsecureRequestIfNeeded.